jena-tdb2/src/main/java/org/apache/jena/tdb2/solver/StageMatchTuple.java

```diff
@@ -68,6 +70,17 @@ private static Iterator<BindingNodeId> access(NodeTupleTable nodeTupleTable, Bin
             iterMatches = x.iterator();
         }
 
+        // Add cancel check.
```

Comment:
A check is needed here for the following query on TDB2:
```
time curl http://localhost:3030/ds --data-urlencode 'query=PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#> SELECT ?x { ?x rdfs:label ?y . ?y rdfs:label ?z }' --data-urlencode 'timeout=1'
```

Against this preloaded [TDB2 DBpedia dataset](https://maven.aksw.org/archiva/#artifact-details-download-content~internal/tdb2.org.aksw.data.text2sparql.2025/dbpedia/1.0.0) (~30GB)